src/lib.rs

```diff
@@ -1108,14 +1108,19 @@ pub enum Expression {
         base: Handle<Expression>,
         index: Handle<Expression>,
     },
-    /// Array access with a known index.
+    /// Access the same types as [`Access`] plus [`Struct`] with a known index.
+    ///
+    /// [`Array`]s must have a [`Constant`] size.
```

Comment:
Is this true? I couldn't find code in the validator that would reject dynamically-sized arrays.

Reply:
The validator [sets a bound of the size of the array](https://github.com/gfx-rs/naga/blob/1aa9154964238af8c692cf521ff90e1f2395e147/src/valid/expression.rs#L223) if its size is constant, but sets it to `!0` if it's not, which somehow evaluates to -1, making the bounds check always fail. At least in theory, I haven't explicitly tested this.

Reply:
It makes the compile time bounds check always _succeed_ because `!0` as a u32 is `u32::MAX` so no index is ever out of bounds. The code could be made easier to grok by using `u32::MAX` explicitly.

Reply:
(Agreed that the code is not as clear as it might be.)